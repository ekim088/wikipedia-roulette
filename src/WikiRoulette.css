.WikiRoulette {
	text-align: center;
}

.WikiRoulette-logo {
	animation: App-logo-spin infinite 20s linear;
	height: 80px;
}

.WikiRoulette-header {
	background-color: #222;
	height: 150px;
	padding: 20px;
	color: white;
}

.WikiRoulette-title {
	font-size: 1.5em;
}

.WikiRoulette-intro {
	font-size: large;
}

@keyframes App-logo-spin {
	from { transform: rotate(0deg); }
	to { transform: rotate(360deg); }
}
