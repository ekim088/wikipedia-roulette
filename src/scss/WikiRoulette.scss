body {
	background-color: #fefcfe;
}

/* .WikiRoulette {
	height: 100%;
	width: 100%;
}

.WikiRoulette .backdrop {
	background-color: #ececec;
	height: 100%;
	left: 0;
	position: absolute;
	top: 0;
	width: 100%;
	z-index: -1;
}

.WikiRoulette .articles {
	height: 80%;
	overflow: hidden;
	padding: 40px 0;
	position: relative;
	top: 50%;
	transform: translateY(-50%);
} */

@keyframes App-logo-spin {
	from { transform: rotate(0deg); }
	to { transform: rotate(360deg); }
}
