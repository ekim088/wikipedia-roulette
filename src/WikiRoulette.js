import React, { Component } from 'react';
import WikiArticle from './WikiArticle';
import './dist/WikiRoulette.css';

export default class WikiRoulette extends Component {
	constructor(props) {
		super(props);
		this.state = {
			articles: [<WikiArticle key="0"/>]
		};
	}

	appendArticle() {
		this.setState({
			articles: [
				...this.state.articles,
				<WikiArticle key={this.state.articles.length}/>
			]
		});
	}

	render() {
		return (
			<div className="WikiRoulette" onClick={() => this.appendArticle()}>
				<div className="backdrop"></div>
				<div className="articles">
					{this.state.articles.slice(0)}
				</div>
			</div>
		);
	}
}
