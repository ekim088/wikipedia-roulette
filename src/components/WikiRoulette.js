import React, { Component } from 'react';
import { connect } from 'react-redux';
import WikiArticle from './WikiArticle';
import '../scss/_WikiRoulette.scss';

class WikiRoulette extends Component {
	constructor(props) {
		super(props);
		this.state = {
			articleComponents: []
		};
	}

	componentDidMount() {
		// load initial article
		this.setState(() => ({
			articleComponents: [<WikiArticle key="0" />]
		}));
	}

	appendArticle() {
		// add new article component
		this.setState(({ articleComponents }) => ({
			articleComponents: [
				...articleComponents,
				<WikiArticle key={articleComponents.length} />
			]
		}));

		// update history component via shared state
		// ...
	}

	render() {
		const { articleComponents } = this.state;

		return (
			<div className="WikiRoulette">
				<div className="backdrop" />
				<pre>{JSON.stringify(this.props)}</pre>
				<button type="button" onClick={() => this.appendArticle()}>
					Add article
				</button>
				<div className="articles">{articleComponents.slice(0)}</div>
			</div>
		);
	}
}

const mapStateToProps = state => ({ ...state });
const mapDispatchToProps = () => ({});

export default connect(mapStateToProps, mapDispatchToProps)(WikiRoulette);
