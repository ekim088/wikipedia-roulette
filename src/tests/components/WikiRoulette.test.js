import React from 'react';
import ReactDOM from 'react-dom';
import { Provider } from 'react-redux';
import configureStore from '../../store';
import WikiRoulette from '../../components/WikiRoulette';

describe('WikiRoulette', () => {
	it('renders without crashing', () => {
		const div = document.createElement('div');
		ReactDOM.render(
			<Provider store={configureStore()}>
				<WikiRoulette />
			</Provider>,
			div
		);
		ReactDOM.unmountComponentAtNode(div);
	});
});
