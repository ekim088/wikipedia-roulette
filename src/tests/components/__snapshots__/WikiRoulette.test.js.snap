// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WikiRoulette renders the app 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(WikiRoulette)>
    <WikiRoulette
      articles={Array []}
    >
      <div
        className="WikiRoulette"
      >
        <div
          className="backdrop"
        />
        <pre>
          {"articles":[]}
        </pre>
        <Cylinder
          articles={
            Array [
              <MountedArticle />,
            ]
          }
        >
          <button
            onClick={[Function]}
            type="button"
          >
            Add article
          </button>
          <section
            className="articles"
          >
            <MountedArticle
              key="0"
            >
              <Connect(WikiArticle)
                callback={[Function]}
              >
                <WikiArticle
                  callback={[Function]}
                  dispatchArticle={[Function]}
                >
                  <article
                    className="wa loading style-0"
                  >
                    <div
                      className="wa-header"
                    >
                      <div
                        className="wa-header-content"
                      />
                    </div>
                    <div
                      className="wa-content"
                    >
                      <a
                        rel="noopener noreferrer"
                        target="_blank"
                      >
                        Test Link
                      </a>
                    </div>
                  </article>
                </WikiArticle>
              </Connect(WikiArticle)>
            </MountedArticle>
          </section>
        </Cylinder>
      </div>
    </WikiRoulette>
  </Connect(WikiRoulette)>
</Provider>
`;
